core/src/window_service.rs

```diff
@@ -211,7 +212,7 @@ where
             debug_assert_matches!(shred, shred::Payload::Shared(_));
         }
         let shred = Shred::new_from_serialized_shred(shred).ok()?;
-        Some((shred, repair))
+        Some((Cow::Owned(shred), repair))
```

Comment:
To confirm, since we're using `Cow` internally, we'd never actually need `Cow::Owned` anywhere right ? Like, we obviously need to make sure the shred don't go out of scope here, but we don't need `Owned` within the `Blockstore` function ?

Reply:
Not sure I understood the questions.
The blockstore insert api don't need owned value but then the recovered shreds will be owned anyways so it has to work with owned values as well as references.
and then also if we only use references, can't always have an iterator over them unless the references are valid for the lifetime of the function, which is not always the case with iterator transformations.